Firestore/Source/Auth/FSTCredentialsProvider.mm

```diff
@@ -67,7 +84,7 @@ - (instancetype)initWithApp:(FIRApp *)app {
   self = [super init];
   if (self) {
     _app = app;
-    _currentUser = [[FSTUser alloc] initWithUID:[self.app getUID]];
+    _currentUser = User(util::MakeStringView([self.app getUID]));
```

Comment:
Given that we can't yet switch to a C++ Firebase App, maybe we should add a constructor to User that hides this incantation. Something like

```c++
#if __OBJC__
explicit User(NSString* uid) : User(util::MakeStringView(uid)) {}
#endif
```

?